stakingModel/test/BasicCalculation.test.js

```diff
@@ -12,144 +12,133 @@ contract("InterestDistribution - Basic calculations", ([user1]) => {
+      await interestDistribution
+        .setITokenToTokenRate(web3.utils.toWei("0.4", "ether"))
+        .catch(e => e);
 
       /**
       * Formula:
-      * GlobalYieldPerToken = GlobalYieldPerToken(P) + [(IntrestTokenRate - LastTokenRate) x IntrestTokenHoldings]/GlobalTotalStake
-      * AvgYieldRatePerToken = [(AvgYieldRatePerToken(P) x TotalStaked) + (AccumulatedYieldPerToken(P) x Staking x (1-%Donation))]/TotalStaked
+      * GlobalYieldPerToken = GlobalYieldPerToken(P) + GDEarnedInterest/GlobalTotalEffectiveStake.
+      * StakeBuyinRate = [StakeBuyinRate(P) + (AccumulatedYieldPerToken(P) x EffectiveStake)]
       * consider :
-      * IntrestTokenRate = 1.02
-      * LastTokenRate = 1
-      * IntrestTokenHoldings = 30 x 1e18
-      * GlobalTotalStake = 20 x 1e18
-      * AccumulatedYieldPerToken(P) = 3
-      * AvgYieldRatePerToken(P) = 0
+      * GlobalTotalEffectiveStake = 20 x 1e18
+      * AccumulatedYieldPerToken(P) = 0
+      * StakeBuyinRate(P) = 0
       * Staking = 10 x 1e18
       * Donation% = 30%
-      * TotalStaked = 20 x 1e18
-      * output = 3
+      * GDEarnedInterest = 10000 x 1e2
       */
 
       await interestDistribution
-        .stakeCalculation(user1, web3.utils.toWei("10", "ether"), 30, web3.utils.toWei("1.02", "ether"), web3.utils.toWei("30", "ether"))
+        .stake(user1, web3.utils.toWei("10", "ether"), 30)
         .catch(e => e);
 
       let yieldData = await interestDistribution.getYieldData(user1);
 
       /**
       * Formula:
-      * GlobalYieldPerToken = GlobalYieldPerToken(P) + [(IntrestTokenRate - LastTokenRate) x IntrestTokenHoldings]/GlobalTotalStake
-      * AvgYieldRatePerToken = [(AvgYieldRatePerToken(P) x TotalStaked) + (AccumulatedYieldPerToken(P) x Staking x (1-%Donation))]/TotalStaked
-      * 0 + [(1.02 - 1) x 30]/20 = [0.02 x 30]x100/20 = 3
-      * (3*10*1e18*(100-30)%)/(20*1e18) = (30*70%)/20 =  1.05 = 1
+      * GlobalYieldPerToken = GlobalYieldPerToken(P) + GDEarnedInterest/GlobalTotalEffectiveStake.
+      * StakeBuyinRate = [StakeBuyinRate(P) + (AccumulatedYieldPerToken(P) x EffectiveStake)]
+      * 0 + (10000)/20 = 500 => 500 x 1e11 (27 + 2(G$ precision) - 18(token decimal) = 11 precision points)
+      * 0 + (500 * 7) = 3500 => 3500 x 1e29 (27 + 2(G$ precision) = 29 precision points)
       */      
-      expect(yieldData[0].toString()).to.be.equal("3");
-      expect(yieldData[1].toString()).to.be.equal("1");
+      expect((yieldData[0]/1e11).toString()).to.be.equal("500");
+      expect((Math.round(yieldData[1]/1e29)).toString()).to.be.equal("3500");
   });
 
-  it("Set Accumulated Yield Per Token and Avg Yield Rate Per Token", async () => {
+  it("Set Accumulated Yield Per Token", async () => {
 
 
     /**
-      * Stake = 5 x 10^18
-      * Donation = 0%
-      * IntrestTokenRate = 1.03
-      * LastTokenRate = 1.02
-      * IntrestTokenHoldings = 60 x 1e18
+      * GDEarnedInterest = 270 x 1e2
+      * interestTokenEarnedToDate = 1.35 x 1e18
```

Comment:
its not interest token earned to date,
its the interest token earned in the current block just like gdearnedinterest.
still i'm not sure the numbers make sense. 
What is  the simulated exchange rate between the interesttoken and G$? does 1.35(X) is  approximately worth 270GD?

Reply:
considering 1cDai = 200 G$